lsfd-cmd/lsfd.c

```diff
@@ -882,13 +882,17 @@ static struct file *collect_file_symlink(struct path_cxt *pc,
 	else if (assoc >= 0) {
 		/* file-descriptor based association */
 		FILE *fdinfo;
+		struct stat lsb;
```

Comment:
I took time to understand the relationship between `lsb` and `sb`.
Instead of adding `lsb` as a local var, adding `bool is_socket` as a local variable looks more straightforward:
```C
	else if (assoc >= 0) {
		/* file-descriptor based association */
		FILE *fdinfo;
		bool is_socket = (sb.st_mode & S_IFMT) == S_IFSOCK;

		if (ul_path_stat(pc, &sb, AT_SYMLINK_NOFOLLOW, name) == 0)
			f->mode = sb.st_mode;

		if (is_nsfs_dev(f->stat.st_dev))
			load_sock_xinfo(pc, name, f->stat.st_ino);

		if (is_socket)
			load_fdsk_xinfo(proc, assoc);
```

Reply:
@masatake thanks!

Fair enough, I think it could be even more simplified, what do you think about the following:
```
                /* file-descriptor based association */
                FILE *fdinfo;
                        
                if (is_nsfs_dev(f->stat.st_dev)) {
                        if (ul_path_stat(pc, &sb, AT_SYMLINK_NOFOLLOW, name) == 0)
                                f->mode = lsb.st_mode;
                        
                        load_sock_xinfo(pc, name, f->stat.st_ino);
                } else if ((sb.st_mode & S_IFMT) == S_IFSOCK) {
                        load_fdsk_xinfo(proc, assoc);
                }
                        
                fdinfo = ul_path_fopenf(pc, "r", "fdinfo/%d", assoc);
                if (fdinfo) {
                        read_fdinfo(f, fdinfo);
                        fclose(fdinfo);
                }
        }    
```
(yet have to test it)

[currently in the process of patching lsns, will update the PR shortly]

Reply:
> @masatake thanks!
> 
> Fair enough, I think it could be even more simplified, what do you think about the following:
> 
> ```
>                 /* file-descriptor based association */
>                 FILE *fdinfo;
>                         
>                 if (is_nsfs_dev(f->stat.st_dev)) {
>                         if (ul_path_stat(pc, &sb, AT_SYMLINK_NOFOLLOW, name) == 0)
>                                 f->mode = lsb.st_mode;
>                         
>                         load_sock_xinfo(pc, name, f->stat.st_ino);
>                 } else if ((sb.st_mode & S_IFMT) == S_IFSOCK) {
>                         load_fdsk_xinfo(proc, assoc);
>                 }
>                         
>                 fdinfo = ul_path_fopenf(pc, "r", "fdinfo/%d", assoc);
>                 if (fdinfo) {
>                         read_fdinfo(f, fdinfo);
>                         fclose(fdinfo);
>                 }
>         }    
> ```
> 
> (yet have to test it)
> 
> [currently in the process of patching lsns, will update the PR shortly]

Actually, we probably need to update `f->mode` for non-namespace fds, so it's not going to work.
I'll use your suggestion with `is_socket`.